### PR TITLE
feat(babel): upgrade to babel 7

### PR DESCRIPTION
**Pull request by imsnif**
This is a fix for: https://github.com/teambit/bit-envs/issues/84, https://github.com/teambit/bit-envs/issues/82 and https://github.com/teambit/bit-envs/issues/50

Please note that projects using babel 6 will likely not work *with their own .babelrc file*. Using the default one or a custom one will likely do the trick. We can create an additional environment that supports babel 6, if there's a need (it's possible to do this in the same repo, but not entirely trivial).

### CODE REVIEW

File: src/babel-dependencies/babel-dependencies.ts [hunk 1]
```diff
@@ -1,80 +1,94 @@
...
+    pluginDeps,
+    presetDeps
+  )
 }
 
-export function getPrefixedPackageName (pluginName: string, prefix: string) {
-  return pluginName.indexOf(prefix) !== 0
-    ? `${prefix}-${pluginName}`
-    : pluginName
+export function getPossiblePkgNames (pkgName: string, pkgType: string) {
```
Comment (GiladShoham) [inline src/babel-dependencies/babel-dependencies.ts]: why not
```js
pkgType: 'plugin' | 'preset'
```

File: src/babel-dependencies/babel-dependencies.ts [hunk 2]
```diff
@@ -1,80 +1,94 @@
...
+  const devDependencies = _get(packageJson, 'devDependencies', {})
+  const peerDependencies = _get(packageJson, 'peerDependencies', {})
+  const packagesInPkgJson = Object.assign({},
+    peerDependencies,
+    devDependencies,
+    dependencies
+  )
+  return names
+    .reduce((pkgNamesAndVersions: any, name: string) => {
+      if (path.isAbsolute(name)) {
```
Comment (GiladShoham) [inline src/babel-dependencies/babel-dependencies.ts]: what about a case which is relative?
```js
plugins = ['../myPlugin']
```
We don't know to handle this as well so we should ignore it for now.

File: src/babel/babel.ts [hunk 3]
```diff
@@ -33,18 +33,25 @@ export function CreateBabelCompiler (name = '.babelrc') {
...
       if (componentDir) {
-        babelrc.plugins = _get(babelrc, 'plugins', []).map(
-          (pluginName: string | Array<string>) => {
-            const actualPluginName = Array.isArray(pluginName)
-              ? pluginName[0]
-              : pluginName
-            return resolvePlugin(componentDir, actualPluginName)
-          }
+        const dependencies = getBabelPackageDependencies(name, info)
+        const findBabelAddons = (babelType: string, addons: Array<string>) => {
```
Comment (GiladShoham) [inline src/babel/babel.ts]: please add a comment about what this function does and why you need it

File: src/babel/babel.ts [hunk 4]
```diff
@@ -33,18 +33,25 @@ export function CreateBabelCompiler (name = '.babelrc') {
...
-
       if (componentDir) {
-        babelrc.plugins = _get(babelrc, 'plugins', []).map(
-          (pluginName: string | Array<string>) => {
-            const actualPluginName = Array.isArray(pluginName)
-              ? pluginName[0]
-              : pluginName
-            return resolvePlugin(componentDir, actualPluginName)
-          }
+        const dependencies = getBabelPackageDependencies(name, info)
```
Comment (GiladShoham) [inline src/babel/babel.ts]: I'm also not sure about this. You should have it in the info. it's already calculated.
I believe in the next extension API it would be much more clear..

File: src/babel-dependencies/babel-dependencies.ts (hunk 1 above)
Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: Cool, I didn't know this was an option for typescript. I tried making it an enum, but there was just so much overhead involved (exporting, importing in different files, etc.) I decided it would just be easier with a comment. :) But this definitely works great.

File: src/babel-dependencies/babel-dependencies.ts (hunk 2 above)
Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: I thought about that, which is why we're indeed ignoring it for now. I didn't want to make it explicit because I was unsure what's the "policy" on bit's side.

File: src/babel/babel.ts (hunk 3 above)
Comment (imsnif) [reply src/babel/babel.ts]: I changed the name of the function (I always prefer that solution if it makes sense). Do you think it's clear enough now? I called it "addons" because I couldn't find the name of the babel entity that is either a "preset" or a "plugin". I figured "addon" would be the most clear. wdyt?

File: src/babel/babel.ts (hunk 4 above)
Comment (imsnif) [reply src/babel/babel.ts]: As we discussed offline, I think this is somewhat of an edge case that is hard to reproduce at the moment because we do not yet test the different environments in a consumer/author setting specifically. I think once we start doing it, a lot of things will come to light.

File: src/babel-dependencies/babel-dependencies.ts [hunk 5]
```diff
@@ -1,80 +1,94 @@
...
+  const babelrc = _get(
+    babelrcFromFind,
+    'config.babel',
+    babelrcFromFind.config
+  )
+  const pluginNames = findBabelPluginNames(babelrc.plugins)
+  const presetNames = babelrc.presets || []
+  const workspaceDir = info.context && info.context.workspaceDir
+  const componentDir = info.context && info.context.componentDir
+  const packageJson = componentDir
```
[the remark this answers is not in the document]
Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: Ah, okay. I think this was me not understanding that function because I was tired. :) What do you think of this change?

File: src/babel-dependencies/babel-dependencies.ts (hunk 2 above)
Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: Now, it would throw a "could not find package for preset/plugin ${name}". Do you think it's better to ignore it?

File: src/babel/babel.ts (hunk 3 above)
Comment (imsnif) [reply src/babel/babel.ts]: What other information would you like to add here?

File: src/babel-dependencies/babel-dependencies.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (GiladShoham) [reply src/babel-dependencies/babel-dependencies.ts]: after the change of loadPackageJsonSync logic, now when you will pass || '' it will eventually run
fs.pathExistsSync('package.json') (and load it)
is this really what we want? 
Maybe add to loadPackageJsonSync if both empty just return undefined?

[the remark this answers is not in the document]
Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: How about now?

File: src/babel-dependencies/babel-dependencies.ts (hunk 2 above)
Comment (GiladShoham) [reply src/babel-dependencies/babel-dependencies.ts]: Good question, i'm not sure about it.
From one hand throw an error forces the user to change it in order to use bit. (And it might not needed for bit, but the user has nothing to do.)
on the other hand - ignoring it might cause unexpected results which is very hard to understand. 
maybe adding some warning about it, or add a config in the raw config tells the compiler to ignore it explicitly. 
what do you think?

Comment (imsnif) [reply src/babel-dependencies/babel-dependencies.ts]: I vote for throwing and adding an ignore feature if the need arises and we still don't support relative paths.
